### PR TITLE
Fix: Fail "helper.saveScreenshot is not a function" on stepByStepReport plugin.

### PR DESCRIPTION
**Pull request by andrerleao**
## Motivation/Description of the PR
- Description of this PR, which problem it solves

Fix: Fail "helper.saveScreenshot is not a function" when using simultaneous:
 - helpers [REST, WebDriver]
 - plugins [autoDelay, stepByStepReport]

Original message fail: <Can't save step screenshot: TypeError: helper.saveScreenshot is not a function> undefined.

Related #2508,  #2513 

Applicable helpers:

- [x] WebDriver
- [ ] Puppeteer
- [ ] Nightmare
- [x] REST
- [ ] FileHelper
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe
- [ ] Playwright

Applicable plugins:

- [ ] allure
- [x] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] puppeteerCoverage
- [ ] retryFailedStep
- [x] screenshotOnFail
- [ ] selenoid
- [x] stepByStepReport
- [ ] wdio

## Type of change

- [ ] :fire: Breaking changes
- [ ] :rocket: New functionality
- [x] :bug: Bug fix
- [ ] :clipboard: Documentation changes/updates
- [x] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [x] Tests have been added
- [x] Documentation has been added (Run `npm run docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Comment by andrerleao** [comment]
If a variable by reference is changed, all other plugins will have their values ​​changed as well. Then, adding 'REST' in autoDelay.js, stepByStepReport.js will have 'REST' to.

I only changed the "supportedHelpers" to variable by value, allowing to change values only in autoDelay.js.

**Comment by PeterNgTr** [comment]
oh, interesting @andrerleao. I didn’t know this 😅 .

**Comment by PeterNgTr** [comment]
Probably we consider to add test for this.

File: lib/plugin/autoDelay.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const store = require('../store');
 const recorder = require('../recorder');
 const event = require('../event');
 const log = require('../output').log;
-const supportedHelpers = require('./standardActingHelpers');
+const supportedHelpers = JSON.parse(JSON.stringify(require('./standardActingHelpers')));
```
Comment (PeterNgTr) [inline lib/plugin/autoDelay.js:6]:
```suggestion
const supportedHelpers = require('./standardActingHelpers').slice();
```

how about using `.slice()` cause it doesn’t change the original array. :)

Comment (andrerleao) [reply lib/plugin/autoDelay.js]: Oh! @PeterNgTr - More clean, now, I didn´t know this ! 😅

**Comment by DavertMik** [comment]
We have removed REST helper from a list so this PR is not needed anymore

**Comment by PeterNgTr** [comment]
hey @DavertMik regarding this `If a variable by reference is changed, all other plugins will have their values ​​changed as well. Then, adding 'REST' in autoDelay.js, stepByStepReport.js will have 'REST' to.` I thought this PR is still needed?

**Comment by andrerleao** [comment]
_> We removed the REST helper from a list, so this PR is no longer needed_

Hello @DavertMik, this is the problem!

In the changes to remove the REST helper, we added this bug :'(

In the autoDelay.js file, the **supoortedHelpers** variable is declared by reference.

> If a variable by reference is changed, all other plug-ins will have their values ​​changed as well, so by adding REST helper to the autoDelay.js file, the plugin, stepByStepReport.js will have 'REST' as well.

In this PR we changed the declaration of the variable to be by **value**, so we changed the list only in this plugin.

Please reconsider this PR.****

**Pull request closed without merging**